File: SWEA/D3/1209. ［S／W 문제해결 기본］ 2일차 － Sum/［S／W 문제해결 기본］ 2일차 － Sum.py
# 테스트케이스 10 고정, 100x100 배열 고정
for _ in range(10):
    t = int(input())
    arr = [list(map(int, input().split())) for _ in range(100)]
    maxA = 0
    maxB = 0
    maxC = 0
     
    # 행, 열, 대각선의 합
    for i in range(100):
        sumA, sumB, sumC = 0, 0, 0
        for j in range(100):
            sumA += arr[i][j]
            sumB += arr[j][i]
            if i == 0:
                sumC += arr[j][j]
        if sumA > maxA: maxA = sumA
        if sumB > maxB: maxB = sumB
        if sumC > maxC: maxC = sumC
     
    # 비교
    result = max(maxA, maxB, maxC)
     
    print(f'#{t} {result}')

File: SWEA/D3/1209. ［S／W 문제해결 기본］ 2일차 － Sum/［S／W 문제해결 기본］ 2일차 － Sum.java
import java.util.*;
import java.io.*;

class Solution
{
	public static void main(String args[]) throws IOException
	{
        //이게 반복문 안에 들어가니 문제가 생겼다.
        //stack overflow가 발생함..ㄷㄷ
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
		for(int T = 0; T < 10; T++) {
            int t = Integer.parseInt(br.readLine());
			
            int[][] arr = new int[100][100];
            
            for(int i = 0; i<100; i++) {
            	StringTokenizer st = new StringTokenizer(br.readLine());
                for (int j = 0; j<100; j++) {
                	arr[i][j] = Integer.parseInt(st.nextToken());
                }
			}
            
            int max = 0;
            
            for (int i = 0; i<100; i++) {
            	int sumA = 0;
                int sumB = 0;
                int sumC = 0;
                sumC += arr[i][i];
                for (int j = 0; j<100; j++) {
                	sumA += arr[i][j];
                    sumB += arr[j][i];
                }
                if (sumA > max) max = sumA;
                if (sumB > max) max = sumB;
                if (sumC > max) max = sumC;
            }
            
            
            System.out.printf("#%d %d\n", t, max);
			
		}// test_case
        
	}//main
    
}//Solution